src/utils/hooks/useNodesRequestParams.ts

```diff
@@ -0,0 +1,49 @@
+import {useEffect, useState} from 'react';
+
+import type {NodesGeneralRequestParams} from '../../store/reducers/nodes/types';
+import type {ProblemFilterValue} from '../../store/reducers/settings/types';
+import {ProblemFilterValues} from '../../store/reducers/settings/settings';
+
+import {HOUR_IN_SECONDS, USE_BACKEND_PARAMS_FOR_TABLES_KEY} from '../constants';
+import {NodesUptimeFilterValues} from '../nodes';
+import {useSetting} from './useSetting';
+
+interface NodesRawRequestParams
+    extends Omit<NodesGeneralRequestParams, 'problems_only' | 'uptime'> {
+    problemFilter?: ProblemFilterValue;
+    nodesUptimeFilter?: NodesUptimeFilterValues;
+}
+
+export const useNodesRequestParams = ({
+    filter,
+    problemFilter,
+    nodesUptimeFilter,
+    sortOrder,
+    sortValue,
+}: NodesRawRequestParams) => {
+    const [useBackendParamsForTables] = useSetting<boolean>(USE_BACKEND_PARAMS_FOR_TABLES_KEY);
+
+    const [requestParams, setRequestParams] = useState<NodesGeneralRequestParams>({});
+
+    // If backend params are enabled, update params value to use them in fetch request
+    // Otherwise no params will be updated, no hooks that depend on requestParams will be triggered
+    useEffect(() => {
```

Comment:
It seems that you don't need `useEffect` and `useState` here.
If `useSetting` rerenders -> `useNodesRequestParams` rerenders and recalculate `request` params in a base flow.

Reply:
It will refetch data on every component rerender (that could be caused not only by request params change).
I replaced two hooks with `useMemo` (additionally also fixed the bug, that params weren't reset on setting toggle)